test/rethinkdb/joins_test.clj

```diff
@@ -0,0 +1,51 @@
+(ns rethinkdb.joins-test
+  (:require [clojure.test :refer :all]
+            [rethinkdb.query :as r]
+            [rethinkdb.test-utils :as u]))
+
+(def names [{:id 1 :name "Pikachu", :type 2},
+            {:id 2 :name "Magnemite", :type [1 2]}])
+(def types [{:id 1 :type "Steel"},
+            {:id 2 :type "Electric"}])
+
+(defn reset [table data]
+  (with-open [conn (r/connect :db u/test-db)]
+    (-> (r/table-drop table)
+        (r/run conn))
+    (u/ensure-table table {} conn)
+    (-> (r/table table)
+        (r/insert data)
+        (r/run conn))))
+
+(defn setup-each [test-fn]
+  (reset :names names)
```

Comment:
`setup-each` should create a `with-open` connection here like `setup-once` does, and pass it to `reset`